compiler-backend/nbe/src/convert.rs

```
@@ -974,10 +1029,129 @@ fn synthesized_evidence(
+            recursive: false,
+            bindings: bindings.into(),
+            body,
+        }))
+    }
+
+    fn shared_evidence(
+        &mut self,
+        evidence: &EvidenceKey,
+    ) -> ConversionResult<Option<ExpressionId>> {
+        let Some(scope) = self.evidence_scopes.last() else { return Ok(None) };
+        let Some(construction) = scope.constructions.get(&evidence).cloned() else {
```

Clippy (1.97.1)

[warning] this expression creates a reference which is immediately dereferenced by the compiler

(warning)